magento-research/pwa-studio
# Graphql playground autodetects queries

_This PR is a cherry-pick from #1198 to make that PR more digestible._

It is number 4 of 6 in the chain of cherry-picked PRs. It needs to be merged _after_ these:

1. #1215
2. #1216 
3. #1217 

It needs to be merged before the next 2:

5. #1224
6. #1198

## Description
Instead of manually supplying query directories to search before starting up the GraphQL playground, this PR hooks into the Webpack compilation to find `.graphql` files and preloads the playground with them.

This makes Venia more portable, since not every Venia-consuming app needs a `queries` directory,

## Related Issue





Factoring this change out of #1198. 

## Verification Steps

1. Add a `.graphql` file anywhere in the project--in a component directory, for instance.
1. Import that file into a component used in the app.
1. Run the dev server (`yarn build && yarn watch:venia`).
1. Open the GraphQL playground to see your `.graphql` file in the preloaded tabs.

## How Have YOU Tested this?
Run devserver for UAT.

## Proposed Labels for Change Type/Package




## Checklist:
